### DISCUSSION TITLE
milvus run error

### DISCUSSION THREAD

**Discussion by w-pq** [milvus-io/milvus#31693] [Q&A and General discussion]
##enviroment
WSL2, compile from source code
##input:
./milvus run standalone
##output:
goroutine 499 [select]:
runtime.gopark(0xc001503f38?, 0x3?, 0x60?, 0xe6?, 0xc001503ef2?)
        /usr/local/go/src/runtime/proc.go:398 +0xce fp=0xc001503d40 sp=0xc001503d20 pc=0x1d1f6ce
runtime.selectgo(0xc001503f38, 0xc001503eec, 0x5c2b220?, 0x0, 0x1?, 0x1)
        /usr/local/go/src/runtime/select.go:327 +0x725 fp=0xc001503e60 sp=0xc001503d40 pc=0x1d30165
google.golang.org/grpc.(*addrConn).resetTransport(0xc0011c4300)
        /home/cheche/gowork/pkg/mod/google.golang.org/grpc@v1.57.0/clientconn.go:1368 +0x3a9 fp=0xc001503f78 sp=0xc001503e60 pc=0x23f5e49
google.golang.org/grpc.(*addrConn).connect(0xc0011c4300)
        /home/cheche/gowork/pkg/mod/google.golang.org/grpc@v1.57.0/clientconn.go:993 +0x99 fp=0xc001503fc8 sp=0xc001503f78 pc=0x23f3d59
google.golang.org/grpc.(*acBalancerWrapper).Connect.func1()
        /home/cheche/gowork/pkg/mod/google.golang.org/grpc@v1.57.0/balancer_conn_wrappers.go:398 +0x25 fp=0xc001503fe0 sp=0xc001503fc8 pc=0x23ee165
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1650 +0x1 fp=0xc001503fe8 sp=0xc001503fe0 pc=0x1d54dc1
created by google.golang.org/grpc.(*acBalancerWrapper).Connect in goroutine 497
        /home/cheche/gowork/pkg/mod/google.golang.org/grpc@v1.57.0/balancer_conn_wrappers.go:398 +0x56
Aborted

**Comment by xiaofan-luan** [comment]
I don't have any experinece to use WSL2. using a docker instead?

**Comment by yhmo** [comment]
Not sure what the error is.

Most of milvus engineers are using this standard workflow to compile and run Milvus locally:

1. Enter the project root path, run this script to set environment paths.
`source ./scripts/setenv.sh`

2. Under the project root path, run this script to install the dependent libraires.
`./scripts/install_deps.sh`


3. Under the project root path, run "make" command to compile milvus.
`make`
Once it successfully compiles, you will see ./bin/milvus under the project root path.


4. Use a text tool to open the ./config/milvus.yaml, replace all the "/var/lib/milvus" to be "/tmp/milvus".
This step is to avoid the permission deny problem during milvus runtime.


5. Start the third-party services by docker, including etcd/minio.
`cd deployments/docker/dev
docker-compose up -d`

6. Back to the project root path, run this script to start a local milvus standalone.
`./scripts/start_standalone.sh`
Use command `ps -e | grep milvus` to check, you will see a milvus process running.

The runtime log will be gernerated under /tmp/standalone.log. If the server successfully starts, you will see log like this:
`---Milvus Proxy successfully initialized and ready to serve!---
[2024/03/21 18:38:57.874 +08:00] [INFO] [components/proxy.go:58] ["Proxy successfully started"]`
